Content.Server/Atmos/Piping/Trinary/EntitySystems/GasMixerSystem.cs

```diff
@@ -205,19 +205,31 @@ private void OnOutputPressureChangeMessage(EntityUid uid, GasMixerComponent mixe
         /// </summary>
         private void OnMixerAnalyzed(EntityUid uid, GasMixerComponent component, GasAnalyzerScanEvent args)
         {
-            if (!EntityManager.TryGetComponent(uid, out NodeContainerComponent? nodeContainer))
-                return;
-
-            var gasMixDict = new Dictionary<string, GasMixture?>();
+            args.GasMixtures ??= new List<(string, GasMixture?)>();
 
-            if(_nodeContainer.TryGetNode(nodeContainer, component.InletOneName, out PipeNode? inletOne))
-                gasMixDict.Add($"{inletOne.CurrentPipeDirection} {Loc.GetString("gas-analyzer-window-text-inlet")}", inletOne.Air);
-            if(_nodeContainer.TryGetNode(nodeContainer, component.InletTwoName, out PipeNode? inletTwo))
-                gasMixDict.Add($"{inletTwo.CurrentPipeDirection} {Loc.GetString("gas-analyzer-window-text-inlet")}", inletTwo.Air);
-            if(_nodeContainer.TryGetNode(nodeContainer, component.OutletName, out PipeNode? outlet))
-                gasMixDict.Add(Loc.GetString("gas-analyzer-window-text-outlet"), outlet.Air);
+            // multiply by volume fraction to make sure to send only the gas inside the analyzed pipe element, not the whole pipe system
+            if (_nodeContainer.TryGetNode(uid, component.InletOneName, out PipeNode? inletOne) && inletOne.Air.Volume != 0f)
```

Comment:
Is there a way to avoid copy/pasting the code from `[GasFilterSystem.cs](https://github.com/space-wizards/space-station-14/pull/25720/files#diff-7e8714bd40c3123783b5481b2a50536ca38c67ef7fedfe147efb3aa37e43ebb6)`?

Reply:
GasFilterSystem.cs and GasMixerSystem.cs are similar in general, but I see no way to combine this without making them both inherit from a common trinary piping entity. I think changing that would be out of scope for this PR, other suggestions are welcome though.

Reply:
Understood, thanks.